### PR TITLE
fix(plugin-cc): event emission fix for different disconnect flows

### PR DESCRIPTION
**Pull request by Kesari3008**
# COMPLETES #AD-HOC

## This pull request addresses
Fixed emitting event from the task object for an event corresponding to CONTACT_ENDED.
Simplified the approach for different disconnect flows.

## by making the following changes

We will emit 'task:end' event for all four disconnect scenarios listed below. This event will send event data as {wrapupRequired: boolean}. 

1. The caller disconnects the call before the agent answers it. wrapupRequired will be set to false.
2. Agent declines the call. wrapupRequired will be set to false.
3. The agent answers the call and ends it. wrapupRequired will be set to true.
4. The agent answers the call and caller disconnects it. wrapupRequired will be set to true.

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [x] Internal code refactor

## The following scenarios were tested
Tested below 4 scenarios:

1. The caller disconnects the call before the agent answers it. 
2. Agent declines the call.
3. The agent answers the call and ends it.
4. The agent answers the call and the caller disconnects it. 

[CC-SDK-COnsole-Logs-Disconnect-Flows.txt.log](https://github.com/user-attachments/files/18200348/CC-SDK-COnsole-Logs-Disconnect-Flows.txt.log)

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

## Release Notes

- **New Features**
	- Introduced a new Contact Center module, enhancing the Webex SDK with contact management capabilities.
	- Added a sample project for the Contact Center, improving user navigation and access to examples.
	- Implemented a comprehensive logging system for better error tracking and reporting.

- **Bug Fixes**
	- Enhanced URL parsing and matching logic for service URLs, improving reliability in service connections.
	- Improved error handling in various components to prevent crashes and ensure smoother user experiences.

- **Documentation**
	- Updated README files and added documentation for the new Contact Center functionalities and usage examples.

- **Tests**
	- Expanded unit test coverage for the Contact Center module, ensuring robust functionality and error handling across various scenarios.

### CODE REVIEW

File: docs/samples/contact-center/app.js [hunk 1]
```diff
@@ -164,7 +164,12 @@ function registerTaskListeners(task) {
   task.on('task:media', (track) => {
     document.getElementById('remote-audio').srcObject = new MediaStream([track]);
   });
-  task.on('task:end', (task) => {
+  task.on('task:end', (wrapupData) => {
+    if (!wrapupData.wrapupRequired) {
+      answerElm.disabled = true;
+      declineElm.disabled = true;
+    }
+    incomingDetailsElm.innerText = '';
```
Comment (sreenara) [inline docs/samples/contact-center/app.js:173]: Please add a console.log here indicating that the task has been ended before it was accepted.

File: packages/@webex/plugin-cc/src/services/task/TaskManager.ts [hunk 2]
```diff
@@ -72,16 +72,20 @@ export default class TaskManager extends EventEmitter {
             this.currentTask = this.currentTask.updateTaskData(payload.data);
             this.currentTask.emit(TASK_EVENTS.TASK_ASSIGNED, this.currentTask);
             break;
+          case CC_EVENTS.AGENT_CONTACT_OFFER_RONA:
           case CC_EVENTS.CONTACT_ENDED:
-            this.emit(TASK_EVENTS.TASK_UNASSIGNED, this.currentTask);
-            if (this.webCallingService.loginOption === LoginOption.BROWSER) {
-              this.currentTask.unregisterWebCallListeners();
-              this.webCallingService.unregisterCallListeners();
+            if (this.currentTask.data.interaction.state === 'new') {
```
Comment (sreenara) [inline packages/@webex/plugin-cc/src/services/task/TaskManager.ts:77]: question. Will we be adding other states in the future? We could make `new` a constant.

**Review by sreenara — CHANGES_REQUESTED** [review]
Changes look good. Minor questions and comments. However, UTs are not added. Please add them and we should be good to go.

File: packages/@webex/plugin-cc/src/services/task/TaskManager.ts (hunk 2 above)
Comment (Kesari3008) [reply packages/@webex/plugin-cc/src/services/task/TaskManager.ts:77]: We may or may not

File: docs/samples/contact-center/app.js (hunk 1 above)
Comment (Kesari3008) [reply docs/samples/contact-center/app.js:173]: added

**Review by adhmenon — APPROVED** [review]

File: packages/@webex/plugin-cc/test/unit/spec/services/task/TaskManager.ts [hunk 3]
```diff
@@ -10,15 +10,6 @@ import { TASK_EVENTS } from '../../../../../src/services/task/types';
 import WebCallingService from '../../../../../src/services/WebCallingService';
 import config from '../../../../../src/config';
 
-jest.mock('./../../../../../src/services/task', () => {
```
Comment (sreenara) [inline packages/@webex/plugin-cc/test/unit/spec/services/task/TaskManager.ts:13]: question. Why have we removed this mock? @Kesari3008

**Review by sreenara — APPROVED** [review]
LGTM. Approving with one question on the UT

File: packages/@webex/plugin-cc/test/unit/spec/services/task/TaskManager.ts (hunk 3 above)
Comment (Kesari3008) [reply packages/@webex/plugin-cc/test/unit/spec/services/task/TaskManager.ts:13]: This is not needed anymore, we are creating actual instance of Task object

**Pull request merged**